### PR TITLE
Require west 0.6.0

### PR DESCRIPTION
**Pull request by mbolivar**
We're starting to cut west 0.6.0 here: https://github.com/zephyrproject-rtos/west/pull/282

The release needs documentation. There are incompatible API changes from v0.5.x, so I can't do that conveniently while still allowing west 0.5.x in zephyr, since the west docs are currently bundled with the Zephyr docs.

So bump the minimum west version and add documentation for the v0.6.x series.

Not mergeable until west 0.6 is out (and until our CI container is updated to install it), but putting this out there for review.

### CODE REVIEW

File: doc/guides/west/release-notes.rst [hunk 1]
```diff
@@ -1,6 +1,91 @@
 West Release Notes
 ##################
 
+v0.6.x
+******
+
+- No separate bootstrapper
+
+  In west v0.5.x, the program was split into two components, a bootstrapper and
```
Comment (carlescufi) [inline doc/guides/west/release-notes.rst]: Maybe you could link to the 1.14 doc here, where all the concepts described in this paragraph are described

File: doc/guides/west/release-notes.rst [hunk 2]
```diff
@@ -1,6 +1,91 @@
...
+  a per-installation clone. The bootstrapper was a program that implemented the
+  :ref:`west init <west-multi-repo-cmds>` command, which cloned the west
+  repository itself along with its other tasks. Except for ``west init``, all
+  other west commands were implemented by code in the cloned west repository,
+  which the bootstrapper delegated to. The bootstrapper itself was distributed
+  via PyPI and installed using ``pip3``.
+
+  This is similar to how Google's Repo tool works, but was determined to be a
+  mistake for west.
```
Comment (carlescufi) [inline doc/guides/west/release-notes.rst]: "mistake" is a strong word. It served its purpose (quick iteration), so I would rephrase this as "west is now stable enough to be entirely distributed through PyPI"

File: doc/guides/west/release-notes.rst [hunk 3]
```diff
@@ -1,6 +1,91 @@
...
+
+  From v0.6.x onwards, all of the core west commands and helper classes are
+  part of the west package distributed via PyPI. This eliminates complexity
+  and makes it possible to import west modules from anywhere in the system,
+  not just extension commands.
+- The ``selfupdate`` command still exists for backwards compatibility, but
+  now simply exits after printing an error message.
+- Manifest syntax changes
```
Comment (carlescufi) [inline doc/guides/west/release-notes.rst]: Maybe mention that the changes are backwards compatible

File: doc/guides/west/repo-tool.rst [hunk 4]
```diff
@@ -152,62 +148,12 @@ The following three source code topologies supported by west:
...
-Per-Installation Clone
-======================
-
-A west installation, as described above, contains a clone of the west
-repository in :file:`.west/west`. This is where the built-in command
-implementations are currently provided. The rest of :ref:`West's APIs
-<west-apis-west>` are also currently provided to extension commands by this
-repository. So that west can update itself, the built-in ``west update`` and
-``west selfupdate`` commands fetch and update the :file:`.west/west` repository.
+This distribution also includes a launcher binary, also named ``west``. When
```
Comment (carlescufi) [inline doc/guides/west/repo-tool.rst]: This is actually not correct on Linux, where the "binary" is actually a Python script

**Review by carlescufi — CHANGES_REQUESTED** [review]
Nice job! Just a few minor nitpicks on the documentation side of things

File: doc/CMakeLists.txt [hunk 5]
```diff
@@ -15,6 +15,11 @@ endif()
 file(TO_CMAKE_PATH "$ENV{ZEPHYR_BASE}" ZEPHYR_BASE)
 
 message(STATUS "Zephyr base: ${ZEPHYR_BASE}")
+if(DEFINED WEST)
```
[the remark this answers is not in the document]
Comment (mbolivar) [reply doc/CMakeLists.txt]: We have it already:

```
-- Found west: /home/mbolivar/.virtualenvs/west-0.6/bin/west (found suitable version "0.6.0", minimum required is "0.5.6")
```

File: doc/guides/west/release-notes.rst (hunk 1 above)
Comment (mbolivar) [reply doc/guides/west/release-notes.rst]: Thanks for bringing up the v1.14 documentation. I'll do that here, and send a separate PR to the v1.14 branch with the following changes as well:

- install west==0.5.8, making note that later versions may work, but it's not guaranteed
- `west init --mr v1.14.1` (since we've already cut v1.14.0, I guess that PR will go out with the next point release)

File: doc/guides/west/release-notes.rst (hunk 3 above)
Comment (mbolivar) [reply doc/guides/west/release-notes.rst]: The syntax is compatible, but the semantics are not (project names must now be unique, which was previously not required), so I don't want to mislead people and would rather not write that here.

File: doc/guides/west/repo-tool.rst (hunk 4 above)
Comment (mbolivar) [reply doc/guides/west/repo-tool.rst]: I will say "launcher executable"

**Comment by mbolivar** [comment]
@carlescufi thanks for review; comments addressed

File: doc/guides/west/build-flash-debug.rst [hunk 6]
```diff
@@ -463,13 +463,22 @@ determined by the imported subclasses of ``ZephyrBinaryRunner``.
 individual runner implementations are in other submodules, such as
 ``runners.nrfjprog``, ``runners.openocd``, etc.
 
-Hacking and APIs
-****************
+Hacking
+*******
+
+This section contains documentation the ``runners.core`` module used by the
```
Comment (dbkinder) [inline doc/guides/west/build-flash-debug.rst]:
```suggestion
This section documents the ``runners.core`` module used by the
```

File: doc/guides/west/release-notes.rst [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+v0.6.x
+******
+
+- No separate bootstrapper
+
+  In west v0.5.x, the program was split into two components, a bootstrapper and
+  a per-installation clone. See `Multiple Repository Management in the v1.14
+  documentation`_ for more details.
+
+  This is similar to how Google's Repo tool works, and let west iterate quickly
```
Comment (dbkinder) [inline doc/guides/west/release-notes.rst]:
```suggestion
  This is similar to how Google's Repo tool works, and lets west iterate quickly
```

File: doc/guides/west/repo-tool.rst [hunk 8]
```diff
@@ -152,62 +148,12 @@ The following three source code topologies supported by west:
...
-
-A west installation, as described above, contains a clone of the west
-repository in :file:`.west/west`. This is where the built-in command
-implementations are currently provided. The rest of :ref:`West's APIs
-<west-apis-west>` are also currently provided to extension commands by this
-repository. So that west can update itself, the built-in ``west update`` and
-``west selfupdate`` commands fetch and update the :file:`.west/west` repository.
+This distribution also includes a launcher executable, also named ``west``. When
+west is installed, the launcher is placed by :file:`pip3` somewhere in the
+user's ``PATH``. This the command-line entry point.
```
Comment (dbkinder) [inline doc/guides/west/repo-tool.rst]:
```suggestion
user's ``PATH``. This is the command-line entry point.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Looks pretty good, thanks!  A few typos found...

**Comment by mbolivar** [comment]
@dbkinder I've applied all your suggestions as-is. Thanks!

File: doc/guides/west/release-notes.rst (hunk 1 above)
Comment (carlescufi) [reply doc/guides/west/release-notes.rst]: Thanks, I've already reviewed the separate PR

**Review by carlescufi — APPROVED** [review]

**Comment by mbolivar** [comment]
Retesting with new CI container that has v0.6.0rc1 installed.

File: doc/getting_started/index.rst [hunk 9]
```diff
@@ -36,7 +36,7 @@ we also have :ref:`documentation for doing without it <no-west>`.)
 Bootstrap west
```
Comment (carlescufi) [inline doc/getting_started/index.rst]: Rename this from Bootstrap to Install

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: doc/getting_started/index.rst (hunk 9 above)
Comment (mbolivar) [reply doc/getting_started/index.rst]: done

**Review by carlescufi — APPROVED** [review]

**Comment by mbolivar** [comment]
Rebased and fixed run_ci.sh now that west can no longer list itself.

File: doc/guides/west/release-notes.rst [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+    URLs directly, like so:
+
+    .. code-block:: yaml
+
+       manifest:
+         projects:
+           - name: example-project-name
+             url: https://github.com/example/example-project
+
+    Project elements with ``url`` attributes set in this way may not also have
```
Comment (vanwinkeljan) [inline doc/guides/west/release-notes.rst]: Would it be an option to add an attribute `remote-name` that can be combined with the attribute `remote`, or the default remote, to form the final URL? The current proposal forces to use the attribute `url` in case two or more projects share the same repo.

As an example for PR #17629 the current west 0.6 implementation would result in:

```
    - name: lvgl_v5
      revision: pull/1/head
      path: modules/lib/gui/lvgl_v5
      url: https://github.com/zephyrproject-rtos/lvgl
    - name: lvgl_v6
      revision: pull/2/head
      path: modules/lib/gui/lvgl_v6
       url: https://github.com/zephyrproject-rtos/lvgl
```
If the `remote-name` attribute would be added this could become:
```
    - name: lvgl_v5
      revision: pull/1/head
      path: modules/lib/gui/lvgl_v5
      remote-name: lvgl
    - name: lvgl_v6
      revision: pull/2/head
      path: modules/lib/gui/lvgl_v6
      remote-name: lvgl
```

Comment (mbolivar) [reply doc/guides/west/release-notes.rst]: > If the remote-name attribute would be added this could become:

How? Sorry, I don't understand the proposal. The `remote` attribute is already the name of the remote, so I'm not sure what `remote-name` means.

```
    - name: lvgl_v5
      revision: pull/1/head
      path: modules/lib/gui/lvgl_v5
      remote-name: lvgl
```

I'm not sure what this is doing; can you please explain? Do you perhaps mean `repo-name` instead of `remote-name`?

> The current proposal forces to use the attribute url in case two or more projects share the same repo.

Right, but what's the issue with that?

Comment (mbolivar) [reply doc/guides/west/release-notes.rst]: Also, would you mind taking this to the west issue tracker? This PR is about documenting the release.

**Comment by mbolivar** [comment]
Kicking CI; there was a timeout (the test passes locally for me with west 0.6.0)

File: doc/guides/west/release-notes.rst (hunk 10 above)
Comment (vanwinkeljan) [reply doc/guides/west/release-notes.rst]: > I'm not sure what this is doing; can you please explain? Do you perhaps mean `repo-name` instead of `remote-name`?

Yes, maybe that is a better naming for it, so `remote` + `repo-name` would form the URL.

> Right, but what's the issue with that?

Redundancy, is this not the reason why the remote concept was introduced?
In case more and more projects are introduced to support different API versions all of them need a hard coded `url`.

> Also, would you mind taking this to the west issue tracker? 

Ofcourse

**Comment by mbolivar** [comment]
Updated docs for 0.6.0rc3 with some additional changes based on feedback from @vanwinkeljan (for which, thanks!).

Once https://github.com/zephyrproject-rtos/ci-dockerfiles/pull/46 is merged and the container is tagged to 0.8rc2, we can run the documentation through CI one last time here, then cut the west release etc.

**Comment by mbolivar** [comment]
Kicking CI again. @dbkinder could you please take another look at the docs? This is ready to merge from the code perspective.

**Review by dbkinder — APPROVED** [review]
docs LGTM, thanks!

**Comment by mbolivar** [comment]
@nashif @MaureenHelm @galak are you good with this too?

File: .shippable.yml [hunk 11]
```diff
@@ -21,7 +21,7 @@ build:
         - ${SHIPPABLE_BUILD_DIR}/ccache
     pre_ci_boot:
         image_name: zephyrprojectrtos/ci
-        image_tag: v0.7
+        image_tag: v0.8.0-rc1
```
Comment (galak) [inline .shippable.yml]: Made a v0.8.0 release with west 0.6.0

**Review by galak — COMMENTED** [review]
Looks good to me, suggest to wait til 0.8.0 dockerimage gets published.

**Comment by carlescufi** [comment]
@galak how far are we to be able to deploy 0.8.0?

File: .shippable.yml (hunk 11 above)
Comment (mbolivar) [reply .shippable.yml]: I've updated this to use the v0.8.0 tag now that I see this is up on docker hub. Ready for this to merge once CI passes?

**Comment by carlescufi** [comment]
Tested with Docker Image 0.8.0
